Get preferred network IP and refactor server package to increase reusability

Samyoul opened the pull request:
### What's New?

- New `GetOutboundIP` function that gets the device's preferred network IP (even works in CI! 🚀 )
  - This is needed for:
    - Generation of TLS certs
    - Creating a TLS listener for network traffic
    - Sharing the network address of the TLS server with other devices
- Restructured TLS certs to be IP address centric, moving away from domain centric
- The TLS media server now hosts on `127.0.0.1`, as `localhost` resolves to `127.0.0.1` any way
- Made all URL generation strictly a `Server` concern
  - No longer need to export `Server.Port`, or assume the server's address outside of the `Server`
  - All consuming code can now be maintained from a single package
- Removed `Server.Port` and replaced with `Server.listener net.Listener`
  - Because the port number is always chosen by the active `net.Listener`, this change just always asks the listener for the port number. If the port number needs to change we change the listener first.
- Implemented `GenerateCertFromKey` for deterministic certs
  - Useful for creating the exact same cert from only a private key
  - Required for `RootCA` whitelisting, which we need.
- Moved server handlers into a modular and extensible format
  - Replaced old Handler structs with simple `HandlerFuncs`
  - Implemented a function `WithHandlers` which allows for multiple sequential calls
- Full TLS server e2e testing via network 👨‍🍳💋
- `Server` config to customise the server

### Note for QA
Potential regression on identicon, image, audio and stickers showing, particularly on iOS. I've tested with emulator and it all works.

---

Required for https://github.com/status-im/status-go/pull/2506

---

TODO
- I would like to remove the use of `globalCertificate *tls.Certificate = nil` and `globalPem string`
  - it seems only useful for `PublicTLSCert`, and globals makes connected functionality hard to trace.

Samyoul committed:
Added function to get preffered network IP

Also done some refactor work oon server package to make a lot more reusable

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?
- [x] Have you tested changes with mobile?
- [x] Have you tested changes with desktop?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (118)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | a803cc4c | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/1/) | 2022-04-05 09:52:31 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220405-095046-a803cc-pr2626.zip) |
| :heavy_check_mark: | a803cc4c | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/1/) | 2022-04-05 09:54:27 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220405-095041-a803cc-pr2626.aar) |
| :heavy_check_mark: | a803cc4c | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/1/) | 2022-04-05 09:54:48 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220405-095041-a803cc-pr2626.zip) |
| | | | | | | |
| :heavy_check_mark: | b59a9529 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/2/) | 2022-04-14 11:37:55 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220414-113555-b59a95-pr2626.zip) |
| :heavy_check_mark: | b59a9529 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/2/) | 2022-04-14 11:38:24 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220414-113553-b59a95-pr2626.zip) |
| :heavy_check_mark: | b59a9529 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/2/) | 2022-04-14 11:40:18 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220414-113553-b59a95-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | 93fc3f14 | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/3/) | 2022-04-14 21:55:03 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220414-215316-93fc3f-pr2626.zip) |
| :heavy_check_mark: | 93fc3f14 | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/3/) | 2022-04-14 21:56:10 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220414-215316-93fc3f-pr2626.zip) |
| :heavy_check_mark: | 93fc3f14 | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/3/) | 2022-04-14 21:57:04 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220414-215316-93fc3f-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | e5836fc4 | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/4/) | 2022-04-14 21:56:51 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220414-215505-e5836f-pr2626.zip) |
| :heavy_check_mark: | e5836fc4 | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/4/) | 2022-04-14 21:57:44 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220414-215612-e5836f-pr2626.zip) |
| :heavy_check_mark: | e5836fc4 | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/4/) | 2022-04-14 21:59:52 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220414-215706-e5836f-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | 8652838a | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/5/) | 2022-04-15 21:32:38 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220415-213105-865283-pr2626.zip) |
| :heavy_check_mark: | 8652838a | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/5/) | 2022-04-15 21:33:07 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220415-213104-865283-pr2626.zip) |
| :heavy_check_mark: | 8652838a | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/5/) | 2022-04-15 21:34:09 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220415-213104-865283-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | 09a0ed0a | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/6/) | 2022-04-16 08:09:59 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220416-080821-09a0ed-pr2626.zip) |
| :heavy_check_mark: | 09a0ed0a | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/6/) | 2022-04-16 08:10:36 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220416-080822-09a0ed-pr2626.zip) |
| :heavy_check_mark: | 09a0ed0a | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/6/) | 2022-04-16 08:11:35 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220416-080821-09a0ed-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | 9395e339 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/7/) | 2022-04-16 08:35:08 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220416-083355-9395e3-pr2626.zip) |
| :heavy_check_mark: | 9395e339 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/7/) | 2022-04-16 08:35:27 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220416-083355-9395e3-pr2626.zip) |
| :heavy_check_mark: | 9395e339 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/7/) | 2022-04-16 08:36:41 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220416-083355-9395e3-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | 8e8b820e | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/8/) | 2022-04-17 14:55:18 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220417-145332-8e8b82-pr2626.zip) |
| :heavy_check_mark: | 8e8b820e | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/8/) | 2022-04-17 15:44:20 | ~50 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220417-145332-8e8b82-pr2626.aar) |
| :heavy_check_mark: | 8e8b820e | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/8/) | 2022-04-17 16:02:30 | ~1 hr 8 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220417-145337-8e8b82-pr2626.zip) |
| | | | | | | |
| :heavy_check_mark: | 49f2170b | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/9/) | 2022-04-17 15:07:25 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220417-150558-49f217-pr2626.zip) |
| :heavy_check_mark: | 49f2170b | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/9/) | 2022-04-17 17:22:17 | ~1 hr 19 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220417-160233-49f217-pr2626.zip) |
| :heavy_check_mark: | 49f2170b | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/9/) | 2022-04-17 17:28:02 | ~1 hr 43 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220417-154423-49f217-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | 8e334d8f | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/10/) | 2022-04-17 19:47:17 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220417-194534-8e334d-pr2626.zip) |
| :heavy_check_mark: | 8e334d8f | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/10/) | 2022-04-17 19:52:27 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220417-194533-8e334d-pr2626.aar) |
| :heavy_check_mark: | 8e334d8f | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/10/) | 2022-04-17 19:52:40 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220417-194536-8e334d-pr2626.zip) |
| | | | | | | |
| :heavy_check_mark: | c1a7bee0 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/11/) | 2022-04-17 20:07:07 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220417-200544-c1a7be-pr2626.zip) |
| :heavy_check_mark: | c1a7bee0 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/11/) | 2022-04-17 20:12:12 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220417-200544-c1a7be-pr2626.zip) |
| :heavy_check_mark: | c1a7bee0 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/11/) | 2022-04-17 20:14:23 | ~8 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220417-200542-c1a7be-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | d558f2bf | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/12/) | 2022-04-19 10:04:42 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220419-100329-d558f2-pr2626.zip) |
| :heavy_check_mark: | d558f2bf | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/12/) | 2022-04-19 10:05:16 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220419-100329-d558f2-pr2626.zip) |
| :heavy_check_mark: | d558f2bf | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/12/) | 2022-04-19 10:06:26 | ~2 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220419-100327-d558f2-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | 1cb2c4df | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/13/) | 2022-04-19 10:56:03 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220419-105402-1cb2c4-pr2626.zip) |
| :heavy_check_mark: | 1cb2c4df | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/13/) | 2022-04-19 10:56:36 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220419-105401-1cb2c4-pr2626.zip) |
| :heavy_check_mark: | 1cb2c4df | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/13/) | 2022-04-19 10:57:56 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220419-105359-1cb2c4-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | 83eb023f | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/14/) | 2022-04-29 12:54:06 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220429-125223-83eb02-pr2626.zip) |
| :heavy_check_mark: | 83eb023f | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/14/) | 2022-04-29 12:55:10 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220429-125220-83eb02-pr2626.zip) |
| :heavy_check_mark: | 83eb023f | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/14/) | 2022-04-29 12:56:45 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220429-125220-83eb02-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | 9e677d38 | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/15/) | 2022-05-03 16:01:23 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220503-155912-9e677d-pr2626.zip) |
| :heavy_check_mark: | 9e677d38 | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/15/) | 2022-05-03 16:01:28 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220503-155912-9e677d-pr2626.zip) |
| :heavy_check_mark: | 9e677d38 | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/15/) | 2022-05-03 16:02:54 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220503-155910-9e677d-pr2626.aar) |
| | | | | | | |
| :x: | 7fd15cd4 | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/16/) | 2022-05-18 13:44:25 | ~1 min | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/16/consoleText) |
| :x: | 7fd15cd4 | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/16/) | 2022-05-18 13:45:09 | ~2 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/16/consoleText) |
| :x: | 7fd15cd4 | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/16/) | 2022-05-18 13:45:21 | ~2 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/16/consoleText) |
| | | | | | | |
| :x: | 078aa222 | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/17/) | 2022-05-18 13:45:13 | ~45 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/17/consoleText) |
| :x: | 078aa222 | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/17/) | 2022-05-18 13:45:58 | ~33 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/17/consoleText) |
| :x: | 078aa222 | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/17/) | 2022-05-18 13:46:05 | ~52 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/17/consoleText) |
| | | | | | | |
| :heavy_check_mark: | 91ff9ddb | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/18/) | 2022-05-18 14:14:02 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220518-140916-91ff9d-pr2626.zip) |
| :heavy_check_mark: | 91ff9ddb | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/18/) | 2022-05-18 14:14:26 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220518-140916-91ff9d-pr2626.zip) |
| :heavy_check_mark: | 91ff9ddb | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/18/) | 2022-05-18 14:16:44 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220518-140916-91ff9d-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | fa0df19f | [#19](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/19/) | 2022-05-18 14:16:25 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220518-141405-fa0df1-pr2626.zip) |
| :heavy_check_mark: | fa0df19f | [#19](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/19/) | 2022-05-18 14:17:33 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220518-141428-fa0df1-pr2626.zip) |
| :heavy_check_mark: | fa0df19f | [#19](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/19/) | 2022-05-18 14:19:48 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220518-141646-fa0df1-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | dc39e31a | [#20](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/20/) | 2022-05-19 12:36:43 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220519-123432-dc39e3-pr2626.zip) |
| :heavy_check_mark: | dc39e31a | [#20](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/20/) | 2022-05-19 12:38:21 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220519-123432-dc39e3-pr2626.zip) |
| :heavy_check_mark: | dc39e31a | [#20](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/20/) | 2022-05-19 12:39:37 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220519-123432-dc39e3-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | a1f42780 | [#21](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/21/) | 2022-05-19 15:46:00 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220519-154335-a1f427-pr2626.zip) |
| :heavy_check_mark: | a1f42780 | [#21](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/21/) | 2022-05-19 15:46:02 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220519-154335-a1f427-pr2626.zip) |
| :heavy_check_mark: | a1f42780 | [#21](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/21/) | 2022-05-19 15:46:33 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220519-154333-a1f427-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | 57c03e5a | [#22](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/22/) | 2022-05-19 20:23:12 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220519-202142-57c03e-pr2626.zip) |
| :heavy_check_mark: | 57c03e5a | [#22](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/22/) | 2022-05-19 20:23:34 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220519-202142-57c03e-pr2626.zip) |
| :heavy_check_mark: | 57c03e5a | [#22](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/22/) | 2022-05-19 20:26:18 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220519-202142-57c03e-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | 96f98400 | [#23](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/23/) | 2022-05-19 22:03:46 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220519-220215-96f984-pr2626.zip) |
| :heavy_check_mark: | 96f98400 | [#23](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/23/) | 2022-05-19 22:04:11 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220519-220215-96f984-pr2626.zip) |
| :heavy_check_mark: | 96f98400 | [#23](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/23/) | 2022-05-19 22:05:37 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220519-220213-96f984-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | ab58b149 | [#24](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/24/) | 2022-05-19 22:24:56 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220519-222333-ab58b1-pr2626.zip) |
| :heavy_check_mark: | ab58b149 | [#24](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/24/) | 2022-05-19 22:25:25 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220519-222333-ab58b1-pr2626.zip) |
| :heavy_check_mark: | ab58b149 | [#24](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/24/) | 2022-05-19 22:26:39 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220519-222333-ab58b1-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | 69d65492 | [#25](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/25/) | 2022-05-26 10:31:44 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220526-102918-69d654-pr2626.zip) |
| :heavy_check_mark: | 69d65492 | [#25](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/25/) | 2022-05-26 10:32:52 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220526-102918-69d654-pr2626.zip) |
| :heavy_check_mark: | 69d65492 | [#25](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/25/) | 2022-05-26 10:33:26 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220526-102918-69d654-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | 3d458b2a | [#26](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/26/) | 2022-05-27 10:01:23 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220527-095925-3d458b-pr2626.zip) |
| :heavy_check_mark: | 3d458b2a | [#26](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/26/) | 2022-05-27 10:01:30 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220527-095927-3d458b-pr2626.zip) |
| :heavy_check_mark: | 3d458b2a | [#26](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/26/) | 2022-05-27 10:02:46 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220527-095924-3d458b-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | a942dd66 | [#27](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/27/) | 2022-05-27 14:54:22 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220527-145241-a942dd-pr2626.zip) |
| :heavy_check_mark: | a942dd66 | [#27](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/27/) | 2022-05-27 14:54:43 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220527-145241-a942dd-pr2626.zip) |
| :heavy_check_mark: | a942dd66 | [#27](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/27/) | 2022-05-27 14:55:53 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220527-145239-a942dd-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | a9b8b7b7 | [#28](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/28/) | 2022-05-27 19:50:29 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220527-194856-a9b8b7-pr2626.zip) |
| :heavy_check_mark: | a9b8b7b7 | [#28](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/28/) | 2022-05-27 19:50:59 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220527-194856-a9b8b7-pr2626.zip) |
| :heavy_check_mark: | a9b8b7b7 | [#28](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/28/) | 2022-05-27 19:52:22 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220527-194854-a9b8b7-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | cd6d6824 | [#29](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/29/) | 2022-05-30 22:15:17 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220530-221339-cd6d68-pr2626.zip) |
| :heavy_check_mark: | cd6d6824 | [#29](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/29/) | 2022-05-30 22:15:47 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220530-221337-cd6d68-pr2626.zip) |
| :heavy_check_mark: | cd6d6824 | [#29](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/29/) | 2022-05-30 22:17:00 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220530-221336-cd6d68-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | d2eec7f7 | [#30](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/30/) | 2022-05-30 22:19:14 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220530-221747-d2eec7-pr2626.zip) |
| :heavy_check_mark: | d2eec7f7 | [#30](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/30/) | 2022-05-30 22:19:32 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220530-221744-d2eec7-pr2626.zip) |
| :heavy_check_mark: | d2eec7f7 | [#30](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/30/) | 2022-05-30 22:20:47 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220530-221744-d2eec7-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | f4b33e34 | [#31](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/31/) | 2022-05-31 06:18:28 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220531-061626-f4b33e-pr2626.zip) |
| :heavy_check_mark: | f4b33e34 | [#31](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/31/) | 2022-05-31 06:18:33 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220531-061626-f4b33e-pr2626.zip) |
| :heavy_check_mark: | f4b33e34 | [#31](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/31/) | 2022-05-31 06:20:02 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220531-061626-f4b33e-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | f190ea08 | [#32](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/32/) | 2022-05-31 10:50:56 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220531-104852-f190ea-pr2626.zip) |
| :heavy_check_mark: | f190ea08 | [#32](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/32/) | 2022-05-31 10:51:33 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220531-104852-f190ea-pr2626.zip) |
| :heavy_check_mark: | f190ea08 | [#32](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/32/) | 2022-05-31 10:52:21 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220531-104852-f190ea-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | a3bde0f7 | [#33](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/33/) | 2022-05-31 11:27:02 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220531-112539-a3bde0-pr2626.zip) |
| :heavy_check_mark: | a3bde0f7 | [#33](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/33/) | 2022-05-31 11:28:30 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220531-112538-a3bde0-pr2626.zip) |
| :heavy_check_mark: | a3bde0f7 | [#33](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/33/) | 2022-05-31 11:29:00 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220531-112538-a3bde0-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | da3d13e4 | [#34](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/34/) | 2022-05-31 13:07:47 | ~8 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220531-125932-da3d13-pr2626.zip) |
| :heavy_check_mark: | da3d13e4 | [#34](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/34/) | 2022-05-31 13:08:00 | ~8 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220531-125932-da3d13-pr2626.zip) |
| :heavy_check_mark: | da3d13e4 | [#34](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/34/) | 2022-05-31 13:11:42 | ~12 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220531-125932-da3d13-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | 4c9e8377 | [#35](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/35/) | 2022-05-31 19:15:08 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220531-191331-4c9e83-pr2626.zip) |
| :heavy_check_mark: | 4c9e8377 | [#35](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/35/) | 2022-05-31 19:15:47 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220531-191331-4c9e83-pr2626.zip) |
| :heavy_check_mark: | 4c9e8377 | [#35](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/35/) | 2022-05-31 19:16:41 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220531-191329-4c9e83-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | b06e3724 | [#36](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/36/) | 2022-06-06 11:54:04 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220606-115150-b06e37-pr2626.zip) |
| :heavy_check_mark: | b06e3724 | [#36](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/36/) | 2022-06-06 11:54:49 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220606-115150-b06e37-pr2626.zip) |
| :heavy_check_mark: | b06e3724 | [#36](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/36/) | 2022-06-06 11:56:35 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220606-115150-b06e37-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | c69e13a3 | [#37](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/37/) | 2022-06-07 09:30:15 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220607-092620-c69e13-pr2626.zip) |
| :heavy_check_mark: | c69e13a3 | [#37](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/37/) | 2022-06-07 09:33:57 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220607-092620-c69e13-pr2626.aar) |
| :heavy_check_mark: | c69e13a3 | [#37](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/37/) | 2022-06-07 09:38:02 | ~11 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220607-092620-c69e13-pr2626.zip) |
| | | | | | | |
| :heavy_check_mark: | d1d2d80e | [#38](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/38/) | 2022-06-07 10:28:38 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220607-102626-d1d2d8-pr2626.zip) |
| :heavy_check_mark: | d1d2d80e | [#38](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/38/) | 2022-06-07 10:29:39 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220607-102624-d1d2d8-pr2626.aar) |
| :heavy_check_mark: | d1d2d80e | [#38](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/38/) | 2022-06-07 10:33:12 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220607-102626-d1d2d8-pr2626.zip) |
| | | | | | | |
| :heavy_check_mark: | 4b06f68e | [#39](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/39/) | 2022-06-07 11:59:32 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220607-115655-4b06f6-pr2626.zip) |
| :heavy_check_mark: | 4b06f68e | [#39](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/39/) | 2022-06-07 12:01:44 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220607-115654-4b06f6-pr2626.aar) |
| | | | | | | |
| :heavy_check_mark: | 6e838921 | [#40](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/40/) | 2022-06-07 12:21:30 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220607-122003-6e8389-pr2626.zip) |
| :heavy_check_mark: | 6e838921 | [#40](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/40/) | 2022-06-07 12:23:39 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220607-122003-6e8389-pr2626.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | ec0c4d36 | [#41](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/41/) | 2022-06-07 12:59:45 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220607-125818-ec0c4d-pr2626.zip) |
| :heavy_check_mark: | ec0c4d36 | [#41](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/41/) | 2022-06-07 13:01:31 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220607-125814-ec0c4d-pr2626.aar) |
| :heavy_check_mark: | ec0c4d36 | [#40](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/40/) | 2022-06-07 13:07:54 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220607-130439-ec0c4d-pr2626.zip) |
| | | | | | | |
| :heavy_check_mark: | 8abbae4d | [#42](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2626/42/) | 2022-06-15 13:02:39 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220615-130048-8abbae-pr2626.zip) |
| :heavy_check_mark: | 8abbae4d | [#41](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2626/41/) | 2022-06-15 13:04:20 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220615-130044-8abbae-pr2626.zip) |
| :heavy_check_mark: | 8abbae4d | [#42](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2626/42/) | 2022-06-15 13:05:09 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220615-130044-8abbae-pr2626.aar) |

Samyoul committed:
Added server.Option and simplified handler funcs

Samyoul committed:
Added serial number deterministically generated from pk

Samyoul committed:
Debugging TLS server connection

Samyoul committed:
Implemented configurable server ip

When accessing over the network the server needs to listen on the network port and not localhost or 127.0.0.1 . Also the cert can now have a dedicated IP

Samyoul commented on protocol/common/message.go:
```diff
@@ -164,17 +161,17 @@ type Message struct {
 	if m.QuotedMessage != nil && m.QuotedMessage.ContentType == int64(protobuf.ChatMessage_IMAGE) {
-		m.QuotedMessage.PrepareImageURL(port)
+		m.QuotedMessage.ImageLocalURL = s.MakeImageURL(m.QuotedMessage.ID)
 	}
 	if m.ContentType == protobuf.ChatMessage_IMAGE {
-		m.ImageLocalURL = fmt.Sprintf("https://localhost:%d/messages/images?messageId=%s", port, m.ID)
+		m.ImageLocalURL = s.MakeImageURL(m.ID)
 	}
 	if m.ContentType == protobuf.ChatMessage_AUDIO {
-		m.AudioLocalURL = fmt.Sprintf("https://localhost:%d/messages/audio?messageId=%s", port, m.ID)
+		m.AudioLocalURL = s.MakeAudioURL(m.ID)
```
Making URLs is `Server`'s concern, `Message` doesn't care.

Samyoul commented on protocol/messenger.go:
```diff
@@ -390,7 +390,8 @@ func NewMessenger(
 
 	mailservers := mailserversDB.NewDB(database)
-	httpServer, err := server.NewServer(database, logger)
+	mediaServer, err := server.NewServer(database, logger)
+	mediaServer.WithMediaHandlers()
```
Handlers need to be attached depending on the server being inited

Samyoul commented on protocol/messenger_images.go:
```diff
@@ -1,7 +1,5 @@
-
 func (m *Messenger) ImageServerURL() string {
-	return fmt.Sprintf("https://localhost:%d/messages/", m.httpServer.Port)
+	return m.httpServer.MakeImageServerURL()
```
URLs are the concern of `Server`, `Messenger` doesn't want to know how they were made

Samyoul committed:
Refactor of URL funcs to use the url package

Samyoul committed:
Removed redundant Options pattern in favour of config param

Samyoul committed:
Added full server test using GetOutboundIP

Samyoul commented on server/handlers.go:
```diff
@@ -0,0 +1,114 @@
+
+type HandlerPatternMap map[string]http.HandlerFunc
+
+func handleIdenticon(logger *zap.Logger) func(w http.ResponseWriter, r *http.Request) {
+	return func(w http.ResponseWriter, r *http.Request) {
```
The three media handlers were originally creating structs that took params, these new `HandlerFunc` funcs take params and return handlers. Less overhead, more extensible, easier to implement

Samyoul commented:
Aww yeah! 🥳 🎉 

```
[2022-04-17T20:13:58.089Z] === RUN   Test_makeSerialNumberFromKey
[2022-04-17T20:13:58.089Z] --- PASS: Test_makeSerialNumberFromKey (0.00s)
[2022-04-17T20:13:58.089Z] === RUN   TestGetOutboundIPWithFullServerE2e
[2022-04-17T20:13:58.089Z] (string) (len=12) "135.181.1.48"
[2022-04-17T20:13:58.089Z] --- PASS: TestGetOutboundIPWithFullServerE2e (0.01s)
[2022-04-17T20:13:58.089Z] === RUN   TestServer_MakeBaseURL
[2022-04-17T20:13:58.089Z] --- PASS: TestServer_MakeBaseURL (0.00s)
[2022-04-17T20:13:58.089Z] === RUN   TestServer_MakeImageServerURL
[2022-04-17T20:13:58.089Z] --- PASS: TestServer_MakeImageServerURL (0.00s)
[2022-04-17T20:13:58.089Z] === RUN   TestServer_MakeIdenticonURL
[2022-04-17T20:13:58.089Z] --- PASS: TestServer_MakeIdenticonURL (0.00s)
[2022-04-17T20:13:58.089Z] === RUN   TestServer_MakeImageURL
[2022-04-17T20:13:58.089Z] --- PASS: TestServer_MakeImageURL (0.00s)
[2022-04-17T20:13:58.089Z] === RUN   TestServer_MakeAudioURL
[2022-04-17T20:13:58.089Z] --- PASS: TestServer_MakeAudioURL (0.00s)
[2022-04-17T20:13:58.089Z] PASS
[2022-04-17T20:13:58.089Z] ok  	github.com/status-im/status-go/server	0.028s
```

bitgamma commented on server/server.go:
```diff
@@ -2,202 +2,86 @@ package server
 		s.logger.Error("failed to start server, retrying", zap.Error(err))
-		s.Port = 0
+		s.port = 0 //TODO find out why the port is set to 0 here
 		err = s.Start()
```
because this error happens when something is already listening on that port. Setting to 0 takes another random port. This happens on iOS when the app is put in background the port is not released for a long time but the socket is closed, so you would enter an infinite loop trying to retake the previous port.

Samyoul replied:
Thank you very much. That is helpful to know, I'll change my code to handle randomly selected port numbers rather than hardcoding a specific port.

Samyoul replied:
@bitgamma I've change the way `Server` works, we no longer use `port` instead we rely on having an active `net.Listener`. This method is explicit that the listener should always choose the port number, we don't need to reset the port number, we just reset the listener. What do you think?

bitgamma replied:
I am ooo (vacation) so I cannot review in details, but in general make sure that the transition from fg/bg and back on both Android and iOS is not broken by this, cause they both have quirks. Also, is there the risk that if you leave an image or a chat open and lock the screen, when you resume the images will all be broken since the port changed. The reason why I used a random port but then tried to keep it was to avoid broken links

Samyoul replied:
Ah sorry, I didn't realise you were on holiday.

> The reason why I used a random port but then tried to keep it was to avoid broken links

Yes, this is what I've done basically. I really like the idea of the `net.Listener` choosing the port number. So instead of storing the port number on the `Server`, I've replaced that with storing the listener itself, as the listener is always the source of the port number.

Samyoul replied:
What I'll do is write some tests to check fg/bg transitioning doesn't produce any errors. My standard e2e test works fine, so that's a good staring point.

bitgamma replied:
ok, but e2e is not going to be enough, especially on iOS sockets are silently killed in some cases. This really needs to be tested on hw

Samyoul committed:
Remove references and usage of Server.port

The application does not need to set the port, we rely on the net.Listener to pick a port.

Samyoul commented:
👍 full e2e server / client test with UDP selected IP and Listener selected port number
```
[2022-04-19T10:10:53.843Z] === RUN   TestGetOutboundIPWithFullServerE2e
[2022-04-19T10:10:53.843Z] (string) (len=26) "https://135.181.5.32:37389"
[2022-04-19T10:10:53.843Z] --- PASS: TestGetOutboundIPWithFullServerE2e (0.12s)
```

Samyoul committed:
Version bump

Samyoul committed:
Added ToECDSA func and improved cert testing

Samyoul committed:
Added error check in test

churik commented:
@Samyoul move it to `Ready for client testing` when it is time to verify it on the client-side. Also if there is any possible regression from your POV, mentioning that in the description would be highly appreciated. Thanks!

Samyoul committed:
Split Server types, embedding raw Server funcs into specialised server types

Samyoul committed:
localhost

Samyoul committed:
Implemented DNS and IP based cert gen

ios doesn't allow for restricted ip addresses to be used in a valid tls cert

pavloburykh commented:
@Samyoul thanx for the PR. Please, take a look at following issue

### ISSUE 1 Images, audio and stickers are not shown both on IOS and Android

**Steps:**
1) Login/register multiaccount
2) Start a chat
3) Send image/sticker/audio message 
4) Observe the result

**Actual result:**
Image/sticker messages are not shown, audio messages can't be played neither for sender nor for receiver. Same issue with images send in timeline.

**Expected result:**
Image/sticker messages are shown, audio messages can be played. 

Tested on builds from status-react PR that was created with status-go "version": "feature/lan_connection_string"
https://github.com/status-im/status-react/pull/13391

logs from Android attached. Bug is reproducible for both IOS and Android
[geth.log](https://github.com/status-im/status-go/files/8755485/geth.log)


https://user-images.githubusercontent.com/97245802/169845163-4791c421-da3c-4d9a-809c-263165022a34.mp4

Samyoul committed:
Replace listener handling with original port store

Also added handlers as a parameter of the Server
